frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java

```diff
@@ -837,6 +850,104 @@ public boolean isSnapshotDisabled() {
         return snapshotDisabled;
     }
 
+    public boolean isTokenExpired(Instant expirationTime) {
+        return !(Instant.now().isBefore(expirationTime));
+    }
+
+    public String generateToken() {
+        byte[] randomBytes = new byte[6];
+        secureRandom.nextBytes(randomBytes);
+        return baseEncoder.encodeToString(randomBytes);
+    }
+
+    public boolean generateKeyFile() throws IOException {
+        String fileData = " ";
+        String absoluteFilePath = getCanonicalPath(".") + "/key_file.txt";
+        keyFileLocation = absoluteFilePath;
+        File file = new File(absoluteFilePath);
+        if (!file.createNewFile() && !file.exists()) {
+            return false;
+        }
+        Integer timeToExpiration = 30; // in minutes
+        fileData =
+                "Management Key: "
+                        + generateToken()
+                        + "\n"
+                        + "Inference Key: "
+                        + generateToken()
+                        + " --- Expiration time: "
+                        + Instant.now().plusSeconds(TimeUnit.MINUTES.toSeconds(timeToExpiration))
+                        + "\n";
+        Files.write(Paths.get("key_file.txt"), fileData.getBytes());
+        return true;
+    }
+
+    public List<String> parseFile(File tsTokenFile) {
```

Comment:
`parseFile` is invoked by `checkTokenAuthorization` which in turn will be called during all inference and management API calls. Reading from file during each of these calls is going to be expensive. Could we potentially maintain the required data, i.e the tokens and expiration time in memory so that we can readily check against them. Also, we can flush this data to the file using say `updateKeyFile` when it is initially generated and when any of the data changes.

Reply:
Changed to maintain the data in memory and no longer need to parse the key file.